src/StreamingDataFrames/README.md

```diff
@@ -1,3 +1,258 @@
+`Application` (`app`).
+
+Your `dataframe` is a pre-defined (declarative) pipeline that you give to your `app`
+to execute. All of your ETL is done here. 
+
+The `app` manages various Kafka-related setup/teardown and message lifecycle 
+(consuming, committing). It processes each message with the `dataframe` you provide it.
+
+<br>
+
+## Current Features
+
+- A simple framework with familiar Pandas interface to ease newcomers into streaming.
+- Easily integrates with other Python frameworks and ML models.
+- Support for basic serialization formats, including JSON (and Quix-specific)
+- Support for basic stateful operations using RocksDB
+- At-least-once Kafka processing guarantees
+- Designed to run and scale resiliently via container orchestration (like Kubernetes)
+- Also easily runs locally for development and debugging
+- Seamless integration with the Quix platform.
+
+Unfamiliar with the Quix platform? We make building and releasing your applications 
+easy by managing an entire infrastructure stack (Kafka, Kubernetes, CI/CD) for you 
+under the hood, with production software best practices in mind, and all backed by Git. 
+
+[Be sure to check us out at quix.io](https://quix.io/) to get up and running with 
+Kafka, fast!
+
+<br>
+
+## Upcoming Improvements
+
+- Stateful Windowing: tumbling, hopping, and sliding windows
+- Group-bys and joins (for merging topics/keys)
+- State recovery based on Kafka changelog topics
+- Support for exactly-once Kafka processing (aka transactions)
+- Other serialization support like Avro and Protobuf support
+- Schema Registry support
+- Multiple input topic support (for non-stateful applications)
+
+<br>
+
+## Want to get Involved?
+
+We strongly believe in the open source model and would love to build a strong 
+community with you, so please join us!
+
+While we are still working on out contribution guidelines, here are some ways you can 
+get involved right now:
+
+- [Join our community mailing list](https://share.hsforms.com/1WWc91mC_SqCvk8xeUWibMw4yjw2)
+- [Check out our project on GitHub](https://github.com/quixio/quix-streams/)
+- [Hang out with us on Slack](https://join.slack.com/t/stream-processing/shared_invite/zt-25lo3wxo7-28hlegDhR2XqyLWjBQynwA)
+
+
+<br><br>
+
+# The Basics
+
+Here's the minimum you need to know to get started with Streaming DataFrames.
+
+More detailed breakdowns can be found in the [documentation folder](./documentation).
+
+## The `Application` class
+
+This is the handler and entrypoint of any `quixstreams` application.
+
+The `Application` class houses all kafka-related configuration, and can generate all 
+necessary objects you will need to get up and running.
+
+At minimum, you'll need to know your Kafka broker address and the consumer group ID you
+wish to use.
+
+You can get started via:
+```python
+from quixstreams import Application
+
+app = Application(
+   broker_address="my_broker_url",
+   consumer_group="my_consumer_group_name",
+)
+```
```

Comment:
I would modify this sample to also print any message it receives. It should be just adding one line of code and with that you already have a "real" working app